src/OrbitGl/TimerTrack.cpp

```diff
@@ -251,6 +251,13 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
 
   for (auto& chain : chains_by_depth) {
     if (!chain) continue;
+    // In order to draw overlaps correctly, we need for every text box to be drawn (current),
+    // its previous and next text box. In order to avoid looking ahead for the next text (which is
+    // error-prone), we are doing just one traversal of the text boxes, while keeping track of the
+    // previous two timers and will draw the previous one, thus the currents iteration value being
+    // the "next" textbox.
+    // Note: This will require us to draw the last timer after the traversal of the text boxes.
+    // Also note: The draw method will take care of nullptr's being passed into.
```

Comment:
I would add "first iteration" or erase directly this "also note"
// Also note: The draw method will take care of nullptr's being passed into (first iteration).